### PR TITLE
Release 4.2.4

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Review by kovrus — APPROVED** [review]
👍

File: docs/appendices/release-notes/4.2.4.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- Fixed an issue that prevented the ``MATCH`` predicate from working in mixed
+  clusters running 4.1.8 and 4.2.
+
+- Fixed an issue that prevented user defined functions in a custom schema from
+  working if used in a generated column expression.
+
+- Fixed an issue that allowed users to use a function in a generated column
+  that didn't fully match the given arguments, leading to a subsequent runtime
+  failure when trying to access tables.
+
```
Comment (mfussenegger) [inline docs/appendices/release-notes/4.2.4.rst:53]: Looks like the following fix on master was not backported:

> - Fixed exposure of the full qualified name of a sub-script column in
  `information_schema.tables.partitioned_by` and
  `pg_catalog.pg_attribute.attname` to use the CrateDB SQL compatible identifier.


Could you check whether that was intentional?

Comment (mkleen) [reply docs/appendices/release-notes/4.2.4.rst:53]: I couldn't find any information on the original pr https://github.com/crate/crate/pull/10413 if this was intentional. Shall I backport it and add it to the 4.2.4 release ?

Comment (mfussenegger) [reply docs/appendices/release-notes/4.2.4.rst:53]: I don't see a reason why it shouldn't be backported, so yes please do so.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**